src/main/scala/pureconfig/ConfigFieldMapping.scala

```diff
@@ -0,0 +1,24 @@
+package pureconfig
+
+trait ConfigFieldMapping[T] {
```

Comment:
@jcazevedo I like the new proposal. I would like to see the code and review it but, if @leifwickland agreed, I would merge it as soon as the code is ready. Do you want to do it now? I can wait to release pureconfig for a couple of days or I can release it now and then release the configuration system in the next version.